src/kemal/exception_handler.cr

```diff
@@ -11,12 +11,26 @@ module Kemal
     rescue ex : Kemal::Exceptions::CustomException
       call_exception_with_status_code(context, ex, context.response.status_code)
     rescue ex : Exception
+      # Use error handler defined for the current exception if it exists
+      return call_exception_with_exception(context, ex, 500) if Kemal.config.error_handlers.has_key?(ex.class)
```

Comment:
This approach won't work with inheritance.

Reply:
Calling `.has_key?` will check for the exception class only.

Reply:
I suppose it should be something like `Kemal.config.error_handlers.any?{ |key, _| ex.class <= key }` in order to match subclasses of `key` as well.

Reply:
That does identify whether the exception inherits from a known exception but shouldn't this also take into account the order of inheritance? For example

```crystal
class GrandParentException < Exception
end

class ParentException < GrandParentException
end

class ChildException < ParentException
end

error Exception do
  "Generic" 
end

error GrandParentException do
  "Grandparent exception"
end

error ParentException do
  "Parent exception"
end

get "/" do
  raise ChildException.new()
end
```

I'll expect that the raised error in the above logic be caught by the handler for `ParentException`.

Is there any way of identify the order of inheritance at runtime?

Reply:
Sure, you *could* sort classes by their inheritance relationship.

I don't think that would be a good idea though. Just following the order of declaration is probably best. `rescue` works the same way: the first matching branch wins, even if a later type would be a closer fit.

Reply:
True. In that case I think this should achieve the desired behavior